### PR TITLE
Download aur pkg

### LINKED ISSUE #4: Git clone
# Implementation

**Using [Git2-rs](https://github.com/rust-lang/git2-rs)**

Given a package allow the cloning of that package from the `AUR` to a given location specified via a path, assume that it's a complete path that doesn't need to have its environment variables expanded.

In addition allow the standard method of downloading the `tar.gz`, but prefer using `git clone`
Put this inside of `aurr::aur::Aur`

add error handling to `aurr::aur::AurError` following the implementation of error handling for `Reqwest`

### PR DESCRIPTION
**Pull request by NickHackman**
Resolves #4 

## Changes
* Added Dependency `git2`
* Parses `.SRCINFO`
* Lots of Test Cases for Clone/Parsing
* Clone Repository
* Example for cloning

### CODE REVIEW

**Comment by NickHackman** [comment]
**Future Notice**: to avoid comments at the end of lines, switch to `(.+)\s*=\s*([^#]+)`

File: src/aur/pkg_cache.rs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+///     ...
+/// ```
+fn parse_srcinfo(path: &Path) -> Result<HashMap<String, Vec<String>>, PkgCacheError> {
+    let mut map: HashMap<String, Vec<String>> = HashMap::new();
+    // Open file, map all errors to PkgCacheError::IoError
+    let file = File::open(path).map_err(|e| PkgCacheError::IOError(e))?;
+    for (i, line) in BufReader::new(file).lines().enumerate() {
+        lazy_static! {
+            // Captures Key value pairs separated by '\s*=\s*', Line is trimmed removing whitespace
+            static ref CAPTURE_KV_PAIRS: Regex = Regex::new(r"(.+)\s*=\s*(.+)").unwrap();
```
Comment (ErikCraig) [inline src/aur/pkg_cache.rs]: From the arch wiki page https://wiki.archlinux.org/index.php/.SRCINFO
"Fields with other names are ignored. Blank lines and comment lines beginning with a hash sign (#) are also ignored. Lines may be indented."
So, the regex should not match lines with a '#' anywhere but the beginning

Comment (NickHackman) [reply src/aur/pkg_cache.rs]: @ErikCraig 
Do we want to properly enforce all the requirements for `.SRCINFO`? As specified [here](https://wiki.archlinux.org/index.php/.SRCINFO#Specification). That's totally achievable, but it'll hurt our speed a little. Maybe separate the stages into a syntax checker and then an actual parser?

Comment (NickHackman) [reply src/aur/pkg_cache.rs]: This is generated by `Makepkg` there's no point in syntax checking other than the basics.

**Pull request merged**